deepmd/dpmodel/output_def.py

```diff
@@ -307,7 +307,23 @@ def apply_operation(var_def: OutputVariableDef, op: OutputVariableOperation) ->
     -------
     int
         The new category of the variable definition.
+
+    Raises
+    ------
+    ValueError
+        If the operation has been applied to the variable definition,
+        and exceed the maximum limitation.
     """
+    if op == OutputVariableOperation.REDU or op == OutputVariableOperation.DERV_C:
+        if check_operation_applied(var_def, op):
+            raise ValueError(f"operation {op} has been applied")
+    elif op == OutputVariableOperation.DERV_R:
+        if check_operation_applied(var_def, OutputVariableOperation.DERV_R):
+            op = OutputVariableOperation.SEC_DERV_R
+            if check_operation_applied(var_def, OutputVariableOperation.SEC_DERV_R):
+                raise ValueError(f"operation {op} has been applied twice")
```

Comment:
```suggestion
    elif op == OutputVariableOperation.DERV_R:
        if check_operation_applied(var_def, OutputVariableOperation.DERV_R):
            raise ValueError(f"operation {op} has been applied")
    elif op = OutputVariableOperation.SEC_DERV_R:
        if not check_operation_applied(var_def, OutputVariableOperation.DERV_R):
            raise ValueError(f"operation OutputVariableOperation.DERV_R should be applied before {op}")
        if check_operation_applied(var_def, OutputVariableOperation.SEC_DERV_R):
            raise ValueError(f"operation {op} has been applied")
```

Reply:
Do you think we should explicitly pass SEC_DERV to this method? Then we need to check whether it has been applied derivatives in `do_derivative`.

Reply:
It seems that the `SEC_DERV` operation should not be exposed to users..